custom_components/myskoda/climate.py

```diff
@@ -217,39 +311,72 @@ def target_temperature(self) -> None | float:  # noqa: D102
                     ):
-                        _LOGGER.info("Air conditioning detected, stopping first.")
+                        _LOGGER.info("%s mode detected, stopping first.", ac.state)
                         await self.coordinator.myskoda.stop_air_conditioning(
                             self.vehicle.info.vin
                         )
-                    _LOGGER.info("Starting Auxiliary heating.")
+                    if ac.state == AirConditioningState.HEATING_AUXILIARY:
+                        _LOGGER.info("%s already running.", ac.state)
+                        return
+
+                    config = AuxiliaryConfig(
+                        target_temperature=self._target_temperature,
+                        duration_in_seconds=self._duration_in_seconds,
+                        start_mode=self._start_mode,
+                        heater_source=self._heater_source,
+                    )
+                    _LOGGER.info("Starting Auxiliary heating [%s]", config)
+
                     await self.coordinator.myskoda.start_auxiliary_heating(
-                        vin=self.vehicle.info.vin,
-                        spin=spin,
-                        config=AuxiliaryConfig(
-                            target_temperature=TargetTemperature(
-                                temperature_value=target_temperature.temperature_value,
-                            ),
-                        ),
+                        vin=self.vehicle.info.vin, spin=spin, config=config
                     )
                 else:
                     _LOGGER.error("Cannot start auxiliary heater: No S-PIN set.")
+            elif hvac_mode == HVACMode.FAN_ONLY:
+                spin = self.coordinator.config.options.get(CONF_SPIN)
+                if spin is not None:
+                    if (
+                        ac.state != AirConditioningState.OFF
+                        and ac.state != AirConditioningState.VENTILATION
+                    ):
+                        _LOGGER.info("%s mode detected, stopping first.", ac.state)
+                        await self.coordinator.myskoda.stop_air_conditioning(
+                            self.vehicle.info.vin
+                        )
+                    if ac.state == AirConditioningState.VENTILATION:
+                        _LOGGER.info("%s already running.", ac.state)
+                        return
+
+                    config = AuxiliaryConfig(
+                        duration_in_seconds=self._duration_in_seconds,
+                        start_mode=AuxiliaryStartMode.VENTILATION,
+                    )
+                    _LOGGER.info("Starting ventilation [%s]", config)
+
+                    await self.coordinator.myskoda.start_auxiliary_heating(
+                        vin=self.vehicle.info.vin, spin=spin, config=config
+                    )
+                else:
+                    _LOGGER.error("Cannot start ventilation: No S-PIN set.")
```

Comment:
nit: general comment, you can often reduce indentation of a lot of code (making it more readable, with fewer lines getting broken up) by reversing the condition and returning early. Obviously only works if there's no more code to execute further down.

E.g. here

```suggestion
                if spin is None:
                    _LOGGER.error("Cannot start ventilation: No S-PIN set.")
                    return
                if (
                    ac.state != AirConditioningState.OFF
                    and ac.state != AirConditioningState.VENTILATION
                ):
                    _LOGGER.info("%s mode detected, stopping first.", ac.state)
                    await self.coordinator.myskoda.stop_air_conditioning(
                        self.vehicle.info.vin
                    )
                if ac.state == AirConditioningState.VENTILATION:
                    _LOGGER.info("%s already running.", ac.state)
                    return

                config = AuxiliaryConfig(
                    duration_in_seconds=self._duration_in_seconds,
                    start_mode=AuxiliaryStartMode.VENTILATION,
                )
                _LOGGER.info("Starting ventilation [%s]", config)

                await self.coordinator.myskoda.start_auxiliary_heating(
                    vin=self.vehicle.info.vin, spin=spin, config=config
                )
```